### PR TITLE
Susi no answer fix

### LINKED ISSUE #141: Error in case susi server does not responds back with any answer
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/susi_server

**Current behavior:**
Currently whenever the response susi server does not contains any answer the code logic fails. For example whenever the body is of the following form:
```
jQuery321024278653839762554_1508317282877({
  "query": "codeheat",
  "count": 1,
  "client_id": "aG9zdF8xNDEuMTAxLjk5LjEyMV8yNjQ4YWJjNA==",
  "query_date": "2017-10-18T09:04:46.328Z",
  "answers": [],
  "answer_date": "2017-10-18T09:04:46.699Z",
  "answer_time": 371,
  "language": "en",
  "session": {"identity": {
    "type": "host",
    "name": "141.101.99.121_2648abc4",
    "anonymous": true
  }}
});
```
The answers array is having no elements in it.
The current code assumes that there is at least one answer present always and that results in the following error:

image

and the susi typing event goes on forever.

**Expected behavior:**
In case the response is not having any answer there should be a proper message being displayed to the user that susi doesn't understand what is being said.

**Steps to reproduce:**
Type `codeheat` in the typing area and send it to susi. The loading will go on forever and there won't be any error message also popping up.

**Related code:**
```
function getResponse(query) {
    loading();
    $.ajax({
        dataType: "jsonp",
        type: "GET",
        url: "https://api.susi.ai/susi/chat.json?timezoneOffset=-300&q=" + query,
        error: function(xhr,textStatus,errorThrown) {   
                console.log(xhr);
                console.log(textStatus);
                console.log(errorThrown);
                loading(false);
            var response = {
                error: true,
                errorText: "Sorry! request could not be made"
            };
            composeSusiMessage(response);
        },
        success: function (data) {
            // This is where the code fails, there should be a check stating whether data.answers exists 
            // and even if it does it should have a length of at least 1. If not then a proper error message 
            // should be displayed
            data.answers[0].actions.map((action) => {
                var response = composeResponse(action,data.answers[0].data);
                loading(false);
                composeSusiMessage(response);
                if(action.type !== data.answers[0].actions[data.answers[0].actions.length-1].type){
                    loading(); //if not last action then create another loading box for susi response
                }
            });
        }
    });
}
```
**Would you like to work on the issue?**

Yes, I'm taking this up

### PR DESCRIPTION
**Pull request by techpool**
Fixes #141 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
In case the SUSI server returns with no answers it used to throw an error and the loading state used to continue infinitely. This fix resolves that and speaks out a predefined message.

image

#### Changes proposed in this pull request:
- Fix when SUSI server returns with no response.

### CODE REVIEW

File: src/script.js [hunk 1]
```diff
@@ -288,14 +288,23 @@ function getResponse(query) {
...
+		            loading(false);
+		            composeSusiMessage(response);
+		            if(action.type !== data.answers[0].actions[data.answers[0].actions.length-1].type){
+		                loading(); //if not last action then create another loading box for susi response
+		            }
+		        });
+        	} else {
+        		var response = composeResponse({
+        			type: "answer",
+        			expression: "SUSI could not find an answer to your question."
```
Comment (gabru-md) [inline src/script.js]: Can we keep it : `No response found`
As this one is a bit long.

Comment (gabru-md) [reply src/script.js]: And it is not necessary that the input is a question. It can be a general statement that maybe susi can produce no response of.

Comment (gabru-md) [inline src/script.js]: Also please squash your commits.

**Comment by ms10398** [comment]
Whats the status @techpool

**Comment by zamhaq** [comment]
Closing this `PR` due to inactivity.

**Pull request closed without merging**